shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -253,13 +259,53 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
       base: Type,
       methods: Ls[MethodDef[Left[Term, Type]]],
       ctor: Ls[Statement],
-      superParameters: Ls[Term]
+      superParameters: Ls[Term],
+      nested: Ls[NuTypeDef],
+      isNested: Bool
+  ): ModuleSymbol = {
+    val symbol = ModuleSymbol(lexicalName, params.sorted, base, methods, ctor, superParameters, nested, isNested)
+    register(symbol)
+    symbol
+  }
+
+  // in DiffTests, we need to rename `TypingUnit` to some other names
+  // because we would not indicate different names manumally
+  def declareTopModule(
+    lexicalName: Str,
+    stmts: Ls[Statement],
+    nuTypes: Ls[NuTypeDef],
+    allowRenaming: Bool = false
```

Comment:
This function will be invoked again in the driver. We should rename top-level modules only if we execute the diff tests because we don't want to indicate the name of the top-level module by ourselves. Anyway, this default parameter has been removed.